spock-spring/src/main/java/org/spockframework/spring/mock/SpockMockPostprocessor.java

```diff
@@ -199,8 +182,8 @@ private void registerSpy(ConfigurableListableBeanFactory beanFactory,
     }
   }
 
-  private Set<String> findCandidateBeans(ConfigurableListableBeanFactory beanFactory,
-                                         MockDefinition mockDefinition) {
+  private String[] findCandidateBeans(ConfigurableListableBeanFactory beanFactory,
```

Comment:
Why the switch to an array? The spring code also still uses a `Set`?

Reply:
I've changed it to make it more consistent with such methods as `determinePrimaryCandidate` but it will be better to do it other way around and migrate both to Set/Collection (following the original code).